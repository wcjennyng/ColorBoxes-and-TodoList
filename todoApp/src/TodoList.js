//component should render the NewTodoForm component
//should render list of Todo components

import React, { useState }  from "react"
import Todo from "./Todo"
import NewTodoForm from "./NewTodoForm"


const TodoList = () => {
    const [todos, setTodos] = useState([])

    //add a new todo
    const add = newTodo => {
        setTodos(todos => [...todos, newTodo])
    }

    //delete a todo by id
    const remove = id => {
        setTodos(todos => todos.filter(todo => todo.id !== id))
    }

    const todoComponents = todos.map(todo => (
        <Todo   
            remove={remove}
            id={todo.id}
            task={todo.task}
        />
    ))

    return (
        <div>
            <NewTodoForm createTodo={add} />
            <ul>{todoComponents}</ul>
        </div>
    )
}

export default TodoList
