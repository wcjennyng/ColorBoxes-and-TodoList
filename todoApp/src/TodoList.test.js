import React from "react"
import { render, fireEvent } from "@testing-library/react"
import TodoList from "./TodoList"

function addTodo(todoList, task = "test") {
    const taskInput = todoList.getByLabelText("Task:")
    fireEvent.change(taskInput, { target: { value: task } })
    const submitButton = todoList.getByText("Add a todo")
    fireEvent.click(submitButton)
}

it("renders without crashing", function () {
    render(<TodoList />);
})

it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
})

it('can add a todo', function() {
    const list = render(<TodoList />)
    addTodo(list)

    expect(list.getByLabelText("Task:")).toHaveValue("")
    expect(list.getByText("X")).toBeInTheDocument()
})

it('can delete a todo', function() {
    const list = render(<TodoList />)
    addTodo(list)

    const removeButton = list.getByText("X")
    fireEvent.click(removeButton)
    expect(removeButton).not.toBeInTheDocument()
})