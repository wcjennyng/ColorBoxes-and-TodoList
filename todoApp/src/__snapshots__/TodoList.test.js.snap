// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Todo List
    </h2>
    <div>
      <form>
        <label
          for="task"
        >
          Task: 
        </label>
        <input
          id="task"
          name="task"
          type="text"
          value=""
        />
        <button>
          Add a todo
        </button>
      </form>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
